### PR TITLE
expression: implement vectorized evaluation for `builtinGreatestIntSig`

### PR DESCRIPTION
**Pull request by tangwz**
### What problem does this PR solve?
Implement vectorized evaluation for builtinGreatestIntSig.
Issue: #12103

### What is changed and how it works?
```
BenchmarkVectorizedBuiltinCompareFunc/builtinGreatestIntSig-VecBuiltinFunc-4                      200000              6433 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinCompareFunc/builtinGreatestIntSig-NonVecBuiltinFunc-4                    30000             42727 ns/op               0 B/op          0 allocs/op
```

### Check List

Tests

 - Unit test

### CODE REVIEW

File: expression/builtin_compare_vec.go [hunk 1]
```diff
@@ -90,3 +90,39 @@ func (b *builtinLeastDecimalSig) vecEvalDecimal(input *chunk.Chunk, result *chun
...
+
+	i64s := result.Int64s()
+	for j := 1; j < len(b.args); j++ {
+		if err := b.args[j].VecEvalInt(b.ctx, input, buf); err != nil {
+			return err
+		}
+		for i := 0; i < n; i++ {
+			if result.IsNull(i) {
+				continue
+			} else if buf.IsNull(i) {
```
Comment (qw4990) [inline expression/builtin_compare_vec.go]: We can use `result.MergeNulls(buf)` here to speed up.

Comment (tangwz) [reply expression/builtin_compare_vec.go]: fix it. PTAL

File: expression/builtin_compare_vec.go [hunk 2]
```diff
@@ -90,3 +90,38 @@ func (b *builtinLeastDecimalSig) vecEvalDecimal(input *chunk.Chunk, result *chun
...
+		if err := b.args[j].VecEvalInt(b.ctx, input, buf); err != nil {
+			return err
+		}
+
+		result.MergeNulls(buf)
+		for i := 0; i < n; i++ {
+			if result.IsNull(i) {
+				continue
+			}
+			v := buf.GetInt64(i)
```
Comment (Reminiscent) [inline expression/builtin_compare_vec.go]: We can reduce the GetInt64 function call by calling the int64s function outside the loop to get the int64 array.

Comment (tangwz) [reply expression/builtin_compare_vec.go]: @Reminiscent PTAL

**Pull request merged**